rust/kernel/sync/guard.rs

```diff
@@ -67,15 +68,18 @@ pub trait Lock {
     /// The type of the data protected by the lock.
     type Inner: ?Sized;
 
+    /// The type of context, if any, that needs to be stored in the guard.
+    type GuardContext = ();
```

Comment:
Please don't introduce additional default associated types. It's unstable.

Reply:
Thanks for the reminder.

Indeed it's better to avoid it here given that it's not something users should have to implement. I'll remove it.